csharp-advanced-topics/HowToUseStringPool/HowtoUseStringPoolApi/StringPoolHelper.cs

```diff
@@ -0,0 +1,223 @@
+﻿using CommunityToolkit.HighPerformance;
+using CommunityToolkit.HighPerformance.Buffers;
+using System;
+using System.Text.RegularExpressions;
+
+namespace HowtoUseStringPoolApi;
+
+public class StringPoolHelper(IConfiguration configuration)
+{
+    private readonly StringPool _myPool = new();
+    private readonly Dictionary<string, string> _cache = [];
+    private readonly List<string> _logger = [];
+    private readonly IConfiguration _configuration = configuration;
+
+    public string Init()
+    {
+        var value1 = _myPool.GetOrAdd("codemaze".AsSpan());
+        var value2 = _myPool.GetOrAdd("codemaze".AsSpan());
+
+        return string.Format("Hash of value1: {0}, value2: {1}",
+             value1.GetHashCode(), value2.GetHashCode());
+    }
+
+    public string SharedInstance(ReadOnlySpan<char> message)
+    {
+        string result;
+        if (_myPool.Size <= 10)
+            result = _myPool.GetOrAdd(message);
+        else
+            result = StringPool.Shared.GetOrAdd(message);
+
+        return result;
+    }
+
+    public bool AddUser(ReadOnlySpan<char> nameSpan, ReadOnlySpan<char> emailSpan)
+    {
+        var prefixSpan = "USER_".AsSpan();
+        var prefix = _myPool.GetOrAdd(prefixSpan);
+        var name = _myPool.GetOrAdd(nameSpan);
+
+        Span<char> combined = stackalloc char[prefixSpan.Length + nameSpan.Length];
```

Comment:
We should not be using stackalloc here, unless we make this a conditional that will stackalloc when the length is small (~ < 128), otherwise will use a pooled array or allocate a new array. Using stackalloc with an unknown/caller provided argument is very dangerous.

Reply:
Corrected